### DISCUSSION TITLE
Return 404 from getEdgeProps

### DISCUSSION THREAD

**Discussion by tj-kev** [flareact/flareact#158] [Ideas]
Next.js has a feature were you can return a 404 from getStaticProps if required. For example the value you wanted wasn't available in a datastore so you wanted a 404 status code returned to the user and search engines. Next.js does this by returning a "notFound: true" parameter with the props.

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

I think a similar feature for Flareact should be added. Or are there any workarounds anyone has already implemented in their own projects which would enable this already?

**Comment by traffisco** [comment]
I'm not aware about something like this in the wild. 
404 http header for the entire request? What happens on soft navigation when props pulled from the page loader?

*shamelessly riding the topic with a related subject*
Sometimes page loader crashes with 500 and not handled, resulting in no navigation thus bad ui. This case should also be handled and considered.

**Reply by tj-kev** [reply]
I think a 404 could be returned on soft navigation also. This would be picked up by the client side and the 404 page rendered instead.

I know there are other threads which have discussed the static/dynamic 404 pages.

I believe this is important for SEO. Especially the hard load 404.

**Reply by traffisco** [reply]
So basically its just changing the status code of the response of the hard load by passing a parameter and continue normal rendering. How it would be implemented on soft navigation? Props are fetched from the client, page already on 200(unless force reload)

**Reply by tj-kev** [reply]
I guess the endpoint returning the props could also return a 404, but would there be any point? As you say, the page has already loaded with a 200.
